dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java

```diff
@@ -882,14 +887,31 @@ private String getEventPagingQuery( EventSearchParams params )
         return sql;
     }
 
-    private String getDataValueQuery()
+    private String getDataValueQuery( EventSearchParams params )
     {
-        String sql = "select pdv.programstageinstanceid as pdv_id, pdv.created as pdv_created, pdv.lastupdated as pdv_lastupdated, "
-            + "pdv.value as pdv_value, pdv.storedby as pdv_storedby, pdv.providedelsewhere as pdv_providedelsewhere, "
-            + "de.uid as de_uid, de.code as de_code " + "from trackedentitydatavalue pdv "
-            + "inner join dataelement de on pdv.dataelementid=de.dataelementid ";
 
-        return sql;
+        StringBuilder sb = new StringBuilder( "select pdv.programstageinstanceid as pdv_id, pdv.created as pdv_created, pdv.lastupdated as pdv_lastupdated, " );
+        sb.append( "pdv.value as pdv_value, pdv.storedby as pdv_storedby, pdv.providedelsewhere as pdv_providedelsewhere, de.uid as de_uid, de.code as de_code, " );
```

Comment:
You are ending on a ", " so if the next if statement it not true, you will get a syntax error (", from")

you should remove the "," and add it to the start of the string in the if-block